clients/src/main/java/org/apache/kafka/clients/admin/DescribeConsumerGroupsResult.java

```diff
@@ -35,9 +34,9 @@
 @InterfaceStability.Evolving
 public class DescribeConsumerGroupsResult {
 
-    private final Map<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>> futures;
+    private final Map<CoordinatorKey, KafkaFuture<ConsumerGroupDescription>> futures;
 
-    public DescribeConsumerGroupsResult(final Map<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>> futures) {
+    public DescribeConsumerGroupsResult(final Map<CoordinatorKey, KafkaFuture<ConsumerGroupDescription>> futures) {
```

Comment:
Is this a source and binary compatible change?

Reply:
I don't think so, but f5d5f65 changed the type arguments too, from `Map<String, KafkaFuture<ConsumerGroupDescription>>` to `Map<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>>`. f5d5f65 hasn't been released yet. If we need to be compatible with the old constructor then we'll have to change the `CoordinatorKey` type argument as well.

Reply:
Yeah, we should definitely fix that. We need to go via a deprecation cycle and remove in 4.0 for released apis.

Reply:
OK, now reverted to `Map<String, KafkaFuture<ConsumerGroupDescription>>`.